### PR TITLE
Feature/compliance year tests

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**

### CODE REVIEW

**Comment by JohnPadgettCivica** [comment]
More tests around the compliance year

File: src/EA.Weee.RequestHandlers.Tests.Unit/Mapping/EvidenceNoteMapTests.cs [hunk 1]
```diff
@@ -48,13 +48,15 @@ public void Map_GivenNote_StandardPropertiesShouldBeMapped()
             var startDate = fixture.Create<DateTime>();
             var endDate = fixture.Create<DateTime>();
             var recipientId = fixture.Create<Guid>();
+            var complianceYear = fixture.Create<short>();
```
Comment (meglenapnovoselska) [inline src/EA.Weee.RequestHandlers.Tests.Unit/Mapping/EvidenceNoteMapTests.cs:51]: is it not an int?

Comment (meglenapnovoselska) [reply src/EA.Weee.RequestHandlers.Tests.Unit/Mapping/EvidenceNoteMapTests.cs:51]: it's a small int in the db! ignore the comment

File: src/EA.Weee.Web.Tests.Unit/ViewModels/ListOfNotesViewModelBaseTests.cs [hunk 2]
```diff
@@ -97,5 +97,71 @@ public void Map_GivenListOfEvidenceNoteData_ShouldReturnMappedData()
...
+            //arrange
+            var notes = fixture.CreateMany<EvidenceNoteData>().ToList();
+            var date = new DateTime(2022, 1, 1);
+
+            //act
+            var result = testClass.Map(notes, date, null);
+
+            //assert
+            result.ManageEvidenceNoteViewModel.SelectedComplianceYear.Should().Be(2022);
+        }
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web.Tests.Unit/ViewModels/ListOfNotesViewModelBaseTests.cs:131]: would you mind adding a test when manageEvidenceNoteViewModel = null && manageEvidenceNoteViewModel.SelectedComplianceYear <0 ?

File: src/EA.Weee.Web.Tests.Unit/ViewModels/ListOfNotesViewModelBaseTests.cs [hunk 3]
```diff
@@ -97,5 +97,71 @@ public void Map_GivenListOfEvidenceNoteData_ShouldReturnMappedData()
...
+            result.ManageEvidenceNoteViewModel.ComplianceYearList.Count().Should().Be(3);
+            result.ManageEvidenceNoteViewModel.ComplianceYearList.ElementAt(0).Should().Be(2022);
+            result.ManageEvidenceNoteViewModel.ComplianceYearList.ElementAt(1).Should().Be(2021);
+            result.ManageEvidenceNoteViewModel.ComplianceYearList.ElementAt(2).Should().Be(2020);
+        }
+
+        [Fact]
+        public void Map_GivenCurrentDateAndManageEvidenceViewModelIsNull_SelectedComplianceYearShouldBeSet()
+        {
+            //arrange
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web.Tests.Unit/ViewModels/ListOfNotesViewModelBaseTests.cs:122]: is that testing the condition when SelectedComplianceYear get the current year, not the compliance year from the view model, it that's yes, would you mind making the years different -2022 and 2020?  so it's obvious it gets the current year.

**Review by meglenapnovoselska — APPROVED** [review]

File: src/EA.Weee.RequestHandlers.Tests.Unit/AatfEvidence/Requests/GetEvidenceNotesByOrganisationRequestTests.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            exception.Should().BeOfType<ArgumentOutOfRangeException>();
+        }
+
+        [Fact]
+        public void GetEvidenceNotesByOrganisationRequest_GivenValues_PropertiesShouldBeSet()
+        {
+            //arrange
+            var organisationId = fixture.Create<Guid>();
+            var statusList = fixture.CreateMany<NoteStatus>().ToList();
+            var complianceYear = fixture.Create<int>();
```
Comment (meglenapnovoselska) [inline src/EA.Weee.RequestHandlers.Tests.Unit/AatfEvidence/Requests/GetEvidenceNotesByOrganisationRequestTests.cs:79]: compliance year is an int in the request, would it matter?  anywhere else is short

File: src/EA.Weee.Web/ViewModels/Shared/ViewEvidenceNoteViewModel.cs [hunk 5]
```diff
@@ -19,7 +19,9 @@ public class ViewEvidenceNoteViewModel : EvidenceNoteViewModel
         public string WasteDisplay => WasteTypeValue.HasValue ? WasteTypeValue.Value.ToDisplayString() : string.Empty;
 
         [DisplayName("Compliance year")]
-        public string ComplianceYearDisplay => StartDate.Value.Year.ToString();
+        public string ComplianceYearDisplay => ComplianceYear.ToString();
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web/ViewModels/Shared/ViewEvidenceNoteViewModel.cs:22]: in the ViewEvidenceNoteViewModelTest.cs I cant see any test referring to that field, would you mind adding some in there? for the display attribute, if it's null, or what will be displayed if it has a value and etc.

**Review by meglenapnovoselska — APPROVED** [review]

File: src/EA.Weee.Web/ViewModels/Shared/ViewEvidenceNoteViewModel.cs (hunk 5 above)
Comment (meglenapnovoselska) [reply src/EA.Weee.Web/ViewModels/Shared/ViewEvidenceNoteViewModel.cs:22]: there is one for the DisplayNameAttribute actually for the compliance year, but add for the other cases

File: src/EA.Weee.Web.Tests.Unit/ViewModels/ListOfNotesViewModelBaseTests.cs (hunk 2 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web.Tests.Unit/ViewModels/ListOfNotesViewModelBaseTests.cs:131]: Not sure I can do that one? When manageEvidenceNoteViewModel  is null I can't set SelectedComplianceYear, unless I'm misunderstanding the test

File: src/EA.Weee.Web.Tests.Unit/ViewModels/ListOfNotesViewModelBaseTests.cs (hunk 3 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web.Tests.Unit/ViewModels/ListOfNotesViewModelBaseTests.cs:125]: added additional checks

File: src/EA.Weee.RequestHandlers.Tests.Unit/AatfEvidence/Requests/GetEvidenceNotesByOrganisationRequestTests.cs (hunk 4 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.RequestHandlers.Tests.Unit/AatfEvidence/Requests/GetEvidenceNotesByOrganisationRequestTests.cs:79]: will update
